bin/src/named.rs

```diff
@@ -271,124 +268,61 @@ async fn load_zone(
     Ok(authority)
 }
 
-// argument name constants for the CLI options
-const QUIET_ARG: &str = "quiet";
-const DEBUG_ARG: &str = "debug";
-const CONFIG_ARG: &str = "config";
-const ZONEDIR_ARG: &str = "zonedir";
-const PORT_ARG: &str = "port";
-const TLS_PORT_ARG: &str = "tls-port";
-const HTTPS_PORT_ARG: &str = "https-port";
-const QUIC_PORT_ARG: &str = "quic-port";
-
-/// Args struct for all options
-#[allow(dead_code)]
-struct Args {
+/// Cli struct for all options managed with clap derive api.
+#[derive(Debug, Parser)]
+#[clap(name = "Trust-DNS named server", version, about)]
+struct Cli {
+    /// Disable INFO messages, WARN and ERROR will remain
+    #[arg(short = 'q', long = "quiet", conflicts_with = "flag_debug")]
     pub(crate) flag_quiet: bool,
+
+    /// Turn on `DEBUG` messages (default is only `INFO`)
+    #[arg(short = 'd', long = "debug", conflicts_with = "flag_quiet")]
     pub(crate) flag_debug: bool,
-    pub(crate) flag_config: String,
-    pub(crate) flag_zonedir: Option<String>,
+
+    /// Path to configuration file of named server,
+    /// by default `/etc/named.toml`
+    #[arg(
+        short = 'c',
+        long = "config",
+        default_value = "/etc/named.toml",
+        value_name = "NAME",
+        value_hint=clap::ValueHint::FilePath,
+    )]
+    pub(crate) flag_config: PathBuf,
+
+    /// Path to the root directory for all zone files,
+    /// see also config toml
+    #[arg(short = 'z', long = "zonedir", value_name = "DIR", value_hint=clap::ValueHint::DirPath)]
+    pub(crate) flag_zonedir: Option<PathBuf>,
+
+    /// Listening port for DNS queries,
+    /// overrides any value in config file
+    #[arg(short = 'p', long = "port", value_name = "PORT")]
     pub(crate) flag_port: Option<u16>,
+
+    /// Listening port for DNS over TLS queries,
+    /// overrides any value in config file
+    #[arg(long = "tls-port", value_name = "TLS-PORT")]
     pub(crate) flag_tls_port: Option<u16>,
```

Comment:
This change a little bit the error presented to the user we get something like:

for `named -p toto`:
```
thread 'main' panicked at 'bad port argument: ParseIntError { kind: InvalidDigit }', bin/src/named.rs:309:36
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

To
```
error: Invalid value "toto" for '--port <PORT>': invalid digit found in string

For more information try '--help'
```